dev/com.ibm.ws.javamail.1.6/src/com/ibm/ws/javamail/internal/injection/MailSessionResourceFactoryBuilder.java

```diff
@@ -94,34 +81,10 @@ public ResourceFactory createResourceFactory(Map<String, Object> props) throws E
         if (!removeExistingConfigurations(filter.toString()))
             throw new IllegalArgumentException(mailSessionID); // internal error, shouldn't ever have been permitted in server.xml
 
-        mailSessionSvcProps.put(MailSessionService.JNDI_NAME, jndiName);
-        if (description != null)
-            mailSessionSvcProps.put(MailSessionService.DESCRIPTION, description);
-        if (storeProtocol != null)
-            mailSessionSvcProps.put(MailSessionService.STOREPROTOCOL, storeProtocol);
-        if (transportProtocol != null)
-            mailSessionSvcProps.put(MailSessionService.TRANSPORTPROTOCOL, transportProtocol);
-        if (host != null)
-            mailSessionSvcProps.put(MailSessionService.HOST, host);
-        if (user != null)
-            mailSessionSvcProps.put(MailSessionService.USER, user);
-        if (password != null)
-            mailSessionSvcProps.put(MailSessionService.PASSWORD, password);
-        if (from != null)
-            mailSessionSvcProps.put(MailSessionService.FROM, from);
-        if (storeProtocolClass != null)
-            mailSessionSvcProps.put(MailSessionService.STOREPROTOCOLCLASSNAME, storeProtocolClass);
-        if (transportProtocolClass != null)
-            mailSessionSvcProps.put(MailSessionService.TRANSPORTPROTOCOLCLASSNAME, transportProtocolClass);
-        if (property != null)
-            mailSessionSvcProps.put(MailSessionService.PROPERTY, property);
-
-        if (application != null) {
-            mailSessionSvcProps.put("application", application);
-            if (module != null) {
-                mailSessionSvcProps.put("module", module);
-                if (component != null)
-                    mailSessionSvcProps.put("component", component);
```

Comment:
... and application, module, and component here.

Reply:
updated